### PR TITLE
MF-201: Fix issue with future dates when adding programs

### PR DESCRIPTION
**Pull request by denniskigen**
https://issues.openmrs.org/browse/MF-201

Fixes issue where the user could be allowed to select a date in the future as the enrollment date in the Date Enrolled input field. 

Rather than fleshing out fully-fledged form validation logic, this PR relies on the [Constraint Validation API](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5/Constraint_validation) and tries to build out some basic [date input](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date#Setting_maximum_and_minimum_dates) validation.

Screenshot 2020-05-12 at 17 56 40

Translations working:

Screenshot 2020-05-15 at 20 29 51

### CODE REVIEW

File: src/widgets/programs/programs-form.component.tsx [hunk 1]
```diff
@@ -228,16 +233,31 @@ export default function ProgramsForm(props: ProgramsFormProps) {
...
                   </svg>
                 </div>
+                <div className={styles.dateError}>
+                  {enrollmentDateRef &&
+                    !enrollmentDateRef?.current?.validity?.valid && (
+                      <span>
+                        <svg className="omrs-icon" role="img">
+                          <use xlinkHref="#omrs-icon-important-notification"></use>
+                        </svg>
+                        Please enter a date that is either on or before today.
```
Comment (FlorianRappl) [inline src/widgets/programs/programs-form.component.tsx]: We should use `<Trans>` here.

File: src/widgets/programs/programs-form.css [hunk 2]
```diff
@@ -77,3 +77,35 @@
 .buttonStylesBorder {
   border-top: 0.125rem solid var(--omrs-color-interaction);
 }
+
+input[type="date"]:invalid {
```
Comment (FlorianRappl) [inline src/widgets/programs/programs-form.css]: This now applies to all invalid date input types. Should we set this in the `styleguide` instead?

**Review by FlorianRappl — COMMENTED** [review]
LGTM so far!

File: src/widgets/programs/programs-form.css (hunk 2 above)
Comment (denniskigen) [reply src/widgets/programs/programs-form.css]: Indeed we should. I've submitted a PR for this https://github.com/openmrs/openmrs-esm-styleguide/pull/75

File: src/widgets/programs/programs-form.component.tsx (hunk 1 above)
Comment (denniskigen) [reply src/widgets/programs/programs-form.component.tsx]: Sure. Working on it.

**Pull request merged**